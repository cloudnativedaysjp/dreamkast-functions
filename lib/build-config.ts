export interface BuildConfig {
  readonly DreamkastApiBaseUrl: string
  readonly Environment: string
  readonly DomainName: string
  readonly HostedZoneID: string
  readonly ZoneName: string
  readonly AccessControlAllowOrigin: string
  readonly GetTracksURL: string
  readonly AWSProfileRegion: string
}
