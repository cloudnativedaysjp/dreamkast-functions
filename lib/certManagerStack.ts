import { Construct } from 'constructs'
import { StackProps, Stack } from 'aws-cdk-lib'
import {
  Certificate,
  CertificateValidation,
  DnsValidatedCertificate,
} from 'aws-cdk-lib/aws-certificatemanager'
import { HostedZone, IHostedZone } from 'aws-cdk-lib/aws-route53'
import { BuildConfig } from './buildConfig'

export class CertManagerStack extends Stack {
  public readonly certificate: Certificate
  public readonly hostedZone: IHostedZone

  constructor(
    scope: Construct,
    id: string,
    props: StackProps,
    buildConfig: BuildConfig,
  ) {
    super(scope, id, props)

    this.hostedZone = HostedZone.fromHostedZoneAttributes(this, 'CNDHostZone', {
      hostedZoneId: buildConfig.HostedZoneID,
      zoneName: buildConfig.ZoneName,
    })

    this.certificate = new DnsValidatedCertificate(this, 'Certificate', {
      domainName: buildConfig.DomainName,
      hostedZone: this.hostedZone,
      validation: CertificateValidation.fromDns(this.hostedZone),
    })
  }
}
