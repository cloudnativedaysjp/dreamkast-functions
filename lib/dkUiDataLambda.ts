import { Construct } from 'constructs'
import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs'
import { BuildConfig } from './buildConfig'
import { tableNameMap } from './statefulStack'
import { PolicyStatement } from 'aws-cdk-lib/aws-iam'

export function newDkUiDataResources(
  scope: Construct,
  buildConfig: BuildConfig,
) {
  const tableNames = tableNameMap(buildConfig.Environment)

  const getDkUiDataFunction = new NodejsFunction(scope, 'getDkUiData', {
    entry: 'src/getDkUiData.ts',
    environment: {
      TABLENAME: tableNames.dkUiData,
    },
  })
  getDkUiDataFunction.addToRolePolicy(
    new PolicyStatement({
      resources: ['arn:aws:dynamodb:*:*:table/*'],
      actions: ['dynamodb:GetItem'],
    }),
  )

  const postDkUiDataFunction = new NodejsFunction(scope, 'patchDkUiData', {
    entry: 'src/postDkUiData.ts',
    environment: {
      TABLENAME: tableNames.dkUiData,
    },
  })
  postDkUiDataFunction.addToRolePolicy(
    new PolicyStatement({
      resources: ['arn:aws:dynamodb:*:*:table/*'],
      actions: ['dynamodb:PutItem'],
    }),
  )

  return {
    getDkUiDataFunction,
    postDkUiDataFunction,
  }
}
