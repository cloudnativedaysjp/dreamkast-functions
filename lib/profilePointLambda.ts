import { Construct } from 'constructs'
import { Stack, StackProps } from 'aws-cdk-lib'
import { PolicyStatement } from 'aws-cdk-lib/aws-iam'
import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs'
import { StatefulStack, tableNameMap } from './statefulStack'
import { BuildConfig } from './buildConfig'

export type ProfilePointProps = StackProps

export function newProfilePointResources(
  scope: Construct,
  id: string,
  props: ProfilePointProps,
  buildConfig: BuildConfig,
  statefulStack: StatefulStack,
) {
  const tableNames = tableNameMap(buildConfig.Environment)

  // Lambda: postProfilePoint

  const postProfilePointFunction = new NodejsFunction(
    scope,
    'postProfilePoint',
    {
      entry: 'src/postProfilePoint.ts',
      environment: {
        TABLENAME: tableNames.profilePoint,
      },
    },
  )
  postProfilePointFunction.addToRolePolicy(
    new PolicyStatement({
      resources: [statefulStack.profilePointTable.tableArn],
      actions: ['dynamodb:PutItem'],
    }),
  )

  // Lambda: GetProfilePoint

  const getProfilePointFunction = new NodejsFunction(scope, 'getProfilePoint', {
    entry: 'src/getProfilePoint.ts',
    environment: {
      PROFILE_POINT_TABLENAME: tableNames.profilePoint,
      POINT_EVENT_TABLENAME: tableNames.pointEvent,
    },
  })
  getProfilePointFunction.addToRolePolicy(
    new PolicyStatement({
      resources: [
        statefulStack.profilePointTable.tableArn,
        statefulStack.pointEventTable.tableArn,
      ],
      actions: ['dynamodb:Query'],
    }),
  )

  return {
    postProfilePointFunction,
    getProfilePointFunction,
  }
}
