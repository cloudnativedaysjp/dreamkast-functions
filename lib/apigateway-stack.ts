import { Construct } from 'constructs';
import { Stack, StackProps, Fn,RemovalPolicy  } from 'aws-cdk-lib';
import { Function, IFunction } from 'aws-cdk-lib/aws-lambda';
import { Effect, PolicyDocument, PolicyStatement, Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
import { Certificate } from 'aws-cdk-lib/aws-certificatemanager';
import { ARecord, IHostedZone, RecordTarget } from 'aws-cdk-lib/aws-route53';
import { ApiGatewayDomain } from 'aws-cdk-lib/aws-route53-targets';
import { LogGroup, RetentionDays } from "aws-cdk-lib/aws-logs";

import * as apigateway from 'aws-cdk-lib/aws-apigateway';

import { ViewerCountSchema, ProfilePointSchema, ProfilePointsSchema, VoteSchema } from './schemas';
import { BuildConfig } from './build-config'

export interface APIGatewayProps extends StackProps {
    readonly certificate: Certificate,
    readonly hostedZone: IHostedZone,
    readonly lambda: {
        readonly voteCFP: IFunction,
        readonly postProfilePoint: IFunction,
        readonly getProfilePoint: IFunction,
    }
}

export class APIGatewayStack extends Stack {
    constructor(scope: Construct, id: string, props: APIGatewayProps, buildConfig: BuildConfig) {
        super(scope, id, props)

        // === [ API Gateway ] === 

        const api = new apigateway.RestApi(this, 'dkFunctionsApi',{
            restApiName: `dk-functions-${buildConfig.Environment}`,
            deployOptions: {
                stageName: 'v1',
                throttlingRateLimit: 60,
                throttlingBurstLimit: 3000,
                dataTraceEnabled: true,
                loggingLevel: apigateway.MethodLoggingLevel.INFO,
                accessLogDestination: new apigateway.LogGroupLogDestination(new LogGroup(this, 'ApiLogGroup', {
                    logGroupName: `${id}-apiGateway`,
                    retention: RetentionDays.ONE_MONTH,
                    removalPolicy: RemovalPolicy.DESTROY,
                })),
                accessLogFormat: apigateway.AccessLogFormat.jsonWithStandardFields(),
            },
        });

        const requestValidator = new apigateway.RequestValidator(this, 'ApiRequestValidator', {
            restApi: api,
            validateRequestParameters: true,
            validateRequestBody: true,
        });

         // Custom Domain
        const domainName = new apigateway.DomainName(this, 'CustomDomain',{
            certificate: props.certificate,
            domainName: buildConfig.DomainName,
            endpointType: apigateway.EndpointType.REGIONAL,
        });
        domainName.addBasePathMapping(api,{
            basePath: '',
        });
        // Not used to follow the dk swagger specifications
        //domainName.addApiMapping(api.deploymentStage,{
        //    basePath: 'api/v1',
        //});

         // A Record
        new ARecord(this, 'APIARecod', {
            zone: props.hostedZone,
            recordName: buildConfig.DomainName,
            target: RecordTarget.fromAlias(new ApiGatewayDomain(domainName))
        });

         // EXECUTION ROLE
        const projectApiExecutionRole = new Role(this, 'ProjectApiExecutionRole', {
            assumedBy: new ServicePrincipal('apigateway.amazonaws.com'),
            inlinePolicies: {
            APIGWLambdaPolicy: new PolicyDocument({
                statements: [
                new PolicyStatement({
                    actions: ['lambda:Invoke*'],
                    effect: Effect.ALLOW,
                    resources: ['arn:aws:lambda:*:607167088920:function:*'],
                }),
                ]
            }),
            }
        });

        /* === [ RESOURCES ] === */
        
        const root = api.root;
        const apiv1 = root.addResource('api').addResource('v1')

        // Dreamkast
        new apigateway.ProxyResource(this,'ProxyResource',{
            parent: apiv1,
            anyMethod: true,
            defaultIntegration: new apigateway.HttpIntegration(`${buildConfig.DreamkastApiBaseUrl}/api/v1/{proxy}`,{
                    proxy: true,
                    httpMethod: 'ANY',
                    options:{
                        credentialsPassthrough: true,
                        requestParameters: {
                            "integration.request.path.proxy": "method.request.path.proxy"
                        },
                    },
                }),
            defaultMethodOptions: {
                requestParameters:{
                    "method.request.path.proxy": true,
                }
            },
        });


        // TRACKS
        const tracks = apiv1.addResource('tracks');
        const trackid = tracks.addResource('{trackId}');
        const viewerCount = trackid.addResource('viewer_count', {
            defaultCorsPreflightOptions: {
                statusCode: 200,
                allowOrigins: apigateway.Cors.ALL_ORIGINS,
            }
        });

        // TALKS
        const talks = apiv1.addResource('talks');
        const talkId = talks.addResource('{talkId}')
        const vote = talkId.addResource('vote');

        // Profile
        const profiles = apiv1.addResource('profile');
        const profileId = profiles.addResource('{profileId}');
        const point = profileId.addResource('point');
        const points = profileId.addResource('points');

        /* === [   MODEL   ] === */

        const viewerCountModel = api.addModel('viewerCountModel',{
            contentType: 'application/json',
            modelName: 'ViewerCount',
            schema: ViewerCountSchema,
        })

        const voteModel = api.addModel('voteModel',{
            contentType: 'application/json',
            modelName: 'Vote',
            schema: VoteSchema,
        })

        const profilePointModel = api.addModel('profilePointModel',{
            contentType: 'application/json',
            modelName: 'ProfilePoint',
            schema: ProfilePointSchema,
        })
        
        const profilePointsModel = api.addModel('profilePointsModel',{
            contentType: 'application/json',
            modelName: 'ProfilePoints',
            schema: ProfilePointsSchema,
        })

        /* === [   ResponseParameters   ] === */

        const CorsResponseParameters = {
            'method.response.header.Access-Control-Allow-Methods': "'GET,POST'",
            'method.response.header.Access-Control-Allow-Origin': buildConfig.AccessControlAllowOrigin,
            'method.response.header.Access-Control-Allow-Headers': "'Content-Type,Authorization,X-Amz-Date,X-Api-Key,X-Amz-Security-Token'",
        }
        const CorsMethodResponseParameters = {
            'method.response.header.Access-Control-Allow-Methods': true,
            'method.response.header.Access-Control-Allow-Origin': true,
            'method.response.header.Access-Control-Allow-Headers': true,
        }

        /* === [   IntegrationResponse   ] === */

        const integrationResponse200 = {
            statusCode: '200',
            responseParameters: CorsResponseParameters,
        }
        const integrationResponse400 = {
            statusCode: '400',
            selectionPattern: 'Error400:.*',
            responseParameters: CorsResponseParameters,
            responseTemplates: {
                'application/json': JSON.stringify({"message":"bad request"}),
            },
        }
        const integrationResponse404 = {
            statusCode: '404',
            selectionPattern: 'Error404:.*',
            responseParameters: CorsResponseParameters,
            responseTemplates: {
                'application/json': JSON.stringify({"message":"not found"}),
            },
        }
        const integrationResponse500 = {
            statusCode: '500',
            selectionPattern: 'Error500:.*',
            responseParameters: CorsResponseParameters,
            responseTemplates: {
                'application/json': JSON.stringify({"message":"internal server error"}),
            },
        }

        /* === [   MethodResponse   ] === */

        const methodResponses200 = {
            statusCode: '200',
            responseParameters: CorsMethodResponseParameters,
            responseModels: {
                'application/json': apigateway.Model.EMPTY_MODEL,
            },
        }
        const methodResponses400 = {
            statusCode: '400',
            responseParameters: CorsMethodResponseParameters,
            responseModels: {
                'application/json': apigateway.Model.ERROR_MODEL,
            },
        }
        const methodResponses404 = {
            statusCode: '404',
            responseParameters: CorsMethodResponseParameters,
            responseModels: {
                'application/json': apigateway.Model.ERROR_MODEL,
            },
        }
        const methodResponses500 = {
            statusCode: '500',
            responseParameters: CorsMethodResponseParameters,
            responseModels: {
                'application/json': apigateway.Model.ERROR_MODEL,
            },
        }

        /* === [   METHODS   ] === */

        // GET /tracks/{trackID}/viewer_count -> GetViewerCountFunction

        const getViewerCountFunctionArn = Fn.importValue(`viewerCountStack-GetViewerCountFunction-Arn-${buildConfig.Environment}`);
        const getViewerCountFunction = Function.fromFunctionArn(this, 'GetViewerCountFunction', getViewerCountFunctionArn);
        viewerCount.addMethod('GET',
            // Integration
            new apigateway.LambdaIntegration( getViewerCountFunction,
                {
                    proxy: false,
                    credentialsRole: projectApiExecutionRole,
                    passthroughBehavior: apigateway.PassthroughBehavior.NEVER,
                    requestTemplates: {
                        'application/json': `{
                            #set($trackId = $util.escapeJavaScript($input.params().get("path").get("trackId")))
                            "trackId": #if($trackId == "" ) 0 #{else} "$trackId" #end
                        }`,
                    },
                    integrationResponses: [
                        integrationResponse200,
                        integrationResponse400,
                        integrationResponse404,
                    ],
                },
            ),
            // MethodOptions
            {
                requestParameters: {
                    "method.request.path.trackId": true,
                },
                requestValidator: requestValidator,
                methodResponses: [
                    {
                        statusCode: '200',
                        responseParameters: CorsMethodResponseParameters,
                        responseModels: {
                            // Need ?
                            'application/json': viewerCountModel,
                        },
                    },
                    methodResponses400,
                    methodResponses404,
                ]
            },
        );

        // POST /talk/{trackID}/vote -> VoteCFP
        vote.addMethod('POST',
            // Integration
            new apigateway.LambdaIntegration( props.lambda.voteCFP,
                {
                    proxy: false,
                    credentialsRole: projectApiExecutionRole,
                    passthroughBehavior: apigateway.PassthroughBehavior.NEVER,
                    requestTemplates: {
                        'application/json': `{
                            "eventAbbr":"$util.escapeJavaScript($input.path('$').eventAbbr)",
                            "talkId": "$util.escapeJavaScript($input.params().get("path").get("talkId"))",
                            "globalIp": "$util.escapeJavaScript($context.identity.sourceIp)"
                        }`,
                    },
                    integrationResponses: [
                        integrationResponse200,
                        integrationResponse400,
                        integrationResponse404,
                    ],
                },
            ),
            // MethodOptions
            {
                requestValidator: requestValidator,
                requestParameters: {
                    "method.request.path.talkId": true,
                },
                requestModels: {
                    'application/json': voteModel,
                },
                methodResponses: [
                    methodResponses200,
                    methodResponses400,
                    methodResponses404,
                ]
            },
        );

        // POST /profiles/{profileId}/point -> PostProfilePointFunction
        point.addMethod('POST',
            // Integration
            new apigateway.LambdaIntegration( props.lambda.postProfilePoint,
                {
                    proxy: false,
                    credentialsRole: projectApiExecutionRole,
                    passthroughBehavior: apigateway.PassthroughBehavior.NEVER,
                    requestTemplates: {
                        'application/json': `{
                            "conference":"$util.escapeJavaScript($input.path('$').eventAbbr)",
                            "profileId":"$util.escapeJavaScript($input.params().get("path").get("profileId"))",
                            "point": "$util.escapeJavaScript($input.path('$').point)",
                            "reasonId":"$util.escapeJavaScript($input.path('$').reasonId)"
                        }`,
                    },
                    integrationResponses: [
                        integrationResponse200,
                        integrationResponse400,
                        integrationResponse404,
                    ],
                },
            ),
            // MethodOptions
            {
                requestValidator: requestValidator,
                requestParameters: {
                    "method.request.path.profileId": true,
                },
                requestModels: {
                    'application/json': profilePointModel,
                },
                methodResponses: [
                    methodResponses200,
                    methodResponses400,
                    methodResponses404,
                    methodResponses500,
                ]
            },
        );

        // GET /profiles/{profileId}/points -> GetProfilePointFunction
        points.addMethod('GET',
        // Integration
        new apigateway.LambdaIntegration( props.lambda.getProfilePoint,
            {
                proxy: false,
                credentialsRole: projectApiExecutionRole,
                passthroughBehavior: apigateway.PassthroughBehavior.NEVER,
                requestTemplates: {
                    'application/json': `{
                        "profileId":"$util.escapeJavaScript($input.params().get("path").get("profileId"))",
                        "conference":"$util.escapeJavaScript($input.params("eventAbbr"))"
                    }`,
                },
                integrationResponses: [
                    integrationResponse200,
                    integrationResponse400,
                    integrationResponse500,
                ],
            },
        ),
        // MethodOptions
        {
            requestValidator: requestValidator,
            requestParameters: {
                "method.request.querystring.eventAbbr": false,
                "method.request.path.profileId": true,
            },
            methodResponses: [
                {
                    statusCode: '200',
                    responseParameters: CorsMethodResponseParameters,
                    responseModels: {
                        'application/json': profilePointsModel,
                    },
                },
                methodResponses400,
                methodResponses500,
            ]
        },
    );
    }
}