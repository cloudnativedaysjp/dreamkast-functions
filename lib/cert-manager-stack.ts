import { Construct } from 'constructs'
import { StackProps, Stack } from 'aws-cdk-lib'
import {
  Certificate,
  CertificateValidation,
  DnsValidatedCertificate,
} from 'aws-cdk-lib/aws-certificatemanager'
import { HostedZone, IHostedZone } from 'aws-cdk-lib/aws-route53'
import { BuildConfig } from './build-config'

export function newCertManagerResources(
  scope: Construct,
  id: string,
  props: StackProps,
  buildConfig: BuildConfig,
) {
  const hostedZone = HostedZone.fromHostedZoneAttributes(scope, 'CNDHostZone', {
    hostedZoneId: buildConfig.HostedZoneID,
    zoneName: buildConfig.ZoneName,
  })

  const certificate = new DnsValidatedCertificate(scope, 'Certificate', {
    domainName: buildConfig.DomainName,
    hostedZone: hostedZone,
    validation: CertificateValidation.fromDns(hostedZone),
  })

  return {
    hostedZone,
    certificate,
  }
}
