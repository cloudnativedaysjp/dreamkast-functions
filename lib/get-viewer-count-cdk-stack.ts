import { Construct } from 'constructs';
import { Stack, StackProps, RemovalPolicy, Fn } from 'aws-cdk-lib';
import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
import { LambdaIntegration, RestApi } from 'aws-cdk-lib/aws-apigateway';
import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
import { BuildConfig } from './build-config'

export class GetViewerCountStack extends Stack {

    constructor(scope: Construct, id: string, props: StackProps, buildConfig: BuildConfig) {
        super(scope, id, props)

        const importedViewerCountTableName = Fn.importValue(`viewerCountTableName-${buildConfig.Environment}`);

        const getViewerCountFunction = new NodejsFunction(this, 'getViewerCount',{
            entry: 'src/get_viewer_count.ts',
            environment: {
                TABLENAME: importedViewerCountTableName,
            },
        });
        
        getViewerCountFunction.addToRolePolicy(new PolicyStatement({
            resources: [
                'arn:aws:dynamodb:*:*:table/*'
            ],
            actions: [
                'dynamodb:GetItem'
            ]
        }));

        const api = new RestApi(this, 'viewerCountApi',{
            restApiName: `viewer-count-${buildConfig.Environment}`,
            deployOptions: {
                stageName: 'v1'
            },
        });

        // ViewerCount Api lambda integration:
        const getViewerCountLambdaIntegration = new LambdaIntegration(getViewerCountFunction);
        const getViewerCountLambdaTrackIdResource = api.root.addResource('{trackId}');
        const getViewerCountLambdaResource = getViewerCountLambdaTrackIdResource.addResource('viewer_count');
        getViewerCountLambdaResource.addMethod('GET', getViewerCountLambdaIntegration);

    }
}

