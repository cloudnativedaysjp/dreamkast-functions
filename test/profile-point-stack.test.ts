import * as cdk from 'aws-cdk-lib'
import { Template } from 'aws-cdk-lib/assertions'
import { ProfilePointStack } from '../lib/profilePointLambda'

test('snapshot test', () => {
  const app = new cdk.App()

  const stack = new ProfilePointStack(app, `stack`, {
    stackName: `test`,
    env: {
      region: 'ap-northeast-1',
    },
  })

  // スタックからテンプレート(JSON)を生成
  const template = Template.fromStack(stack).toJSON()

  // 生成したテンプレートとスナップショットが同じか検証
  expect(template).toMatchSnapshot()
})
