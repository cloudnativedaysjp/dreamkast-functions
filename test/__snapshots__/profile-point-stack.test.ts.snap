// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "TableCD117FA1": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "profileId",
            "AttributeType": "N",
          },
          {
            "AttributeName": "conference#timestamp",
            "AttributeType": "S",
          },
        ],
        "BillingMode": "PAY_PER_REQUEST",
        "KeySchema": [
          {
            "AttributeName": "profileId",
            "KeyType": "HASH",
          },
          {
            "AttributeName": "conference#timestamp",
            "KeyType": "RANGE",
          },
        ],
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "getProfilePoint58F6A9AB": {
      "DependsOn": [
        "getProfilePointServiceRoleDefaultPolicyD399569A",
        "getProfilePointServiceRoleEC991F3A",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-ap-northeast-1",
          },
          "S3Key": "[HASH REMOVED]",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "TABLENAME": {
              "Ref": "TableCD117FA1",
            },
          },
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "getProfilePointServiceRoleEC991F3A",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "getProfilePointServiceRoleDefaultPolicyD399569A": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "dynamodb:Query",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "TableCD117FA1",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "getProfilePointServiceRoleDefaultPolicyD399569A",
        "Roles": [
          {
            "Ref": "getProfilePointServiceRoleEC991F3A",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "getProfilePointServiceRoleEC991F3A": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "postProfilePointEFB04901": {
      "DependsOn": [
        "postProfilePointServiceRoleDefaultPolicyF57921C5",
        "postProfilePointServiceRole0F5C54F5",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-ap-northeast-1",
          },
          "S3Key": "[HASH REMOVED]",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "TABLENAME": {
              "Ref": "TableCD117FA1",
            },
          },
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "postProfilePointServiceRole0F5C54F5",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "postProfilePointServiceRole0F5C54F5": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "postProfilePointServiceRoleDefaultPolicyF57921C5": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "dynamodb:PutItem",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "TableCD117FA1",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "postProfilePointServiceRoleDefaultPolicyF57921C5",
        "Roles": [
          {
            "Ref": "postProfilePointServiceRole0F5C54F5",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
